stix2/sources/__init__.py

```diff
@@ -191,6 +256,87 @@ def query(self, query=None):
 
         """
 
+    def relationships(self, obj, relationship_type=None, source_only=False, target_only=False):
+        """Retrieve Relationships involving the given STIX object.
+
+        Only one of `source_only` and `target_only` may be `True`.
+
+        Args:
+            obj (STIX object OR dict OR str): The STIX object (or its ID) whose
+                relationships will be looked up.
+            relationship_type (str): Only retrieve Relationships of this type.
+            source_only (bool): Only retrieve Relationships for which this
+                object is the source_ref. Default: False.
+            target_only (bool): Only retrieve Relationships for which this
+                object is the target_ref. Default: False.
+
+        Returns:
+            (list): List of Relationship objects involving the given STIX object.
+
+        """
+        results = []
+        filters = [Filter('type', '=', 'relationship')]
+
+        try:
+            obj_id = obj.get('id', '')
```

Comment:
If the dictionary doesn't have an `'id'`, do we want to raise an error quicker? Will having the empty string cause problems down the line (either an error, which would be bad, or invalid results, which would also be bad, or just zero results, which would just be inefficient)?